l1-contracts/src.ts/deploy-utils.ts

```diff
@@ -55,3 +55,48 @@ export async function deployViaCreate2(
 
   return [expectedAddress, tx.hash];
 }
+
+export async function deployBytecodeViaCreate2(
+  deployWallet: ethers.Wallet,
+  contractName: string,
+  bytecode: ethers.BytesLike,
+  create2Salt: string,
+  ethTxOptions: ethers.providers.TransactionRequest,
+  create2FactoryAddress: string,
+  verbose: boolean = true
+): Promise<[string, string]> {
+  // [address, txHash]
+
+  const log = (msg: string) => {
+    if (verbose) {
+      console.log(msg);
+    }
+  };
+  log(`Deploying ${contractName}`);
+
+  const create2Factory = SingletonFactoryFactory.connect(create2FactoryAddress, deployWallet);
```

Comment:
this function is almost identical to the one above, it is better to remove the amount of code duplication